entities/weapons/rvr_held_item/shared.lua

```diff
@@ -42,4 +45,53 @@ end
 
 -- Empty to remove default behaviour, don't remove >:(
 function SWEP:PrimaryAttack()
+    self:SetNextPrimaryFire( CurTime() + self.Cooldown )
+
+    if CLIENT then
+        if CurTime() < ( self.nextFire or 0 ) then return end
+        self.nextFire = CurTime() + self.Cooldown
+    end
+
+    local itemType = self:GetItemType()
+    if not itemType then return end
+
+    local itemData = RVR.Items.getItemData( itemType )
+
+    if itemData and itemData.consumable then
+        if not itemData.canConsume( self.Owner ) then return end
+
+        local owner = self.Owner
+
+        hook.Add( "RVR_Inventory_CanChangeHotbarSelected", "RVR_HeldItemConsume" .. owner:EntIndex(), function( ply, idx )
+            if ply ~= owner then return end
+            return false
+        end )
+
+        timer.Simple( self.Cooldown, function()
+            hook.Remove( "RVR_Inventory_CanChangeHotbarSelected", "RVR_HeldItemConsume" .. owner:EntIndex() )
+        end )
+
+        if SERVER then
+            timer.Simple( self.Cooldown, function()
+                local inv = owner.RVR_Inventory
+                local slotData = RVR.Inventory.getSlot( owner, inv.HotbarSelected )
+
+                slotData.count = slotData.count - 1
+                if slotData.count == 0 then
+                    slotData = nil
+                end
+
+                RVR.Inventory.setSlot( owner, inv.HotbarSelected, slotData, { owner } )
```

Comment:
could this be in its own named function, perhaps a method on the SWEP. or a function in RVR.Inventory?

Reply:
Certainly, and would defo be useful elsewhere